msobooking.yaml

```diff
@@ -0,0 +1,473 @@
+entityDefinitions:
+  jailstay:
+    fqn: "ol.jailstay"
+    entitySetName: "MSOJailstay"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
+        column: "SYSID"
+      ol.releasedatetime:
+        type: "ol.releasedatetime"
+        column: "RELDATE"
+      criminaljustice.releasetype:
+        type: "criminaljustice.releasetype"
+        column: "RELTYPE"
+      ol.admissiontype:
+        type: "ol.admissiontype"
+        column: "ADTYPE"
+      ol.commitdatetime:
+        type: "ol.commitdatetime"
+        column: "COMDATE"
+      ol.datasource:
+        type: "ol.datasource"
+        column: "DATASOURCE"    
+    name: "jailstay"
+
+  person:
+    fqn: "general.person"
+    entitySetName: "MSOInmates"
+    propertyDefinitions:
+      nc.SubjectIdentification:
+        type: "nc.SubjectIdentification"
+        column: "PCP"
+      nc.PersonGivenName:
+        type: "nc.PersonGivenName"
+        column: "FIRSTNAM"
+      nc.PersonSurName:
+        type: "nc.PersonSurName"
+        column: "LASTNAME"
+      nc.SSN:
+        type: "nc.SSN"
+        column: "SSNUM"
+      nc.PersonSex:
+        type: "nc.PersonSex"
+        column: "SEX"
+      nc.PersonSuffix:
+        type: "nc.PersonSuffix"
+        column: "AFFIX"
+      nc.PersonBirthDate:
+        type: "nc.PersonBirthDate"
+        column: "BIRTH"
+      nc.PersonEthnicity:
+        type: "nc.PersonEthnicity"
+        column: "HISPANIC"
+      nc.PersonRace:
+        type: "nc.PersonRace"
+        column: "RACE"
+      nc.PersonMiddleName:
+        type: "nc.PersonMiddleName"
+        column: "MIDDLE"
+    name: "person"
+
+  fbinumber:
+    fqn: "ol.identification"
+    entitySetName: "MSOFBINumbers"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
+        column: "FBI"
+      ol.stringnumber:
+        type:  "ol.stringnumber"
+        column: "FBI"
+    name: "fbinumber"
+
+  temporaryrelease:
+    fqn: "ol.temporaryrelease"
+    entitySetName: "MSOTemporaryrelease"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
+        column: "RELSTDT"
+      ol.datetimestart:
+        type: "ol.datetimestart"
+        column: "RELSTDT"
+      ol.datetimeend:
+        type: "ol.datetimeend"
+        column: "RELRNDT"
+      ol.projectedenddatetime:
+        type: "ol.projectedenddatetime"
+        column: "RELDBDATE"
+    name: "temporaryrelease"
+
+  releaseofficer:
+    fqn: "ol.employee"
+    entitySetName: "MSOOfficer"
+    propertyDefinitions:
+      ol.id:
+        type: "ol.id"
+        column: "RELOFF"
+      publicsafety.employeeid:
+        type: "publicsafety.employeeid"
+        column: "RELOFF"
+    name: "releaseofficer"
+
+  dnasample:
+    fqn: "ol.medicaltest"
+    entitySetName: "MSODnaSample"
+    propertyDefinitions:
+      general.id:
+        type: "general.id"
+        column: "DNA_SAMPLE_STATUS"
+      ol.status:
+        type: "ol.status"
+        column: "DNA_SAMPLE_STATUS"
+    name: "dnasample"
+
+  sentence:
```

Comment:
Bringing up the SENTENCE has made me realize that this is also a sparsely populated column - this probably wouldn't have worked on the full integration for that reason as well. I will just use comdate for the PK and write SENTENCE in the notes. Does a column only integrate if the PK exists?  This makes me think that it might be a common mistake to assume that a column has full coverage over the rows which the properties in that entity come from. If so, I'll take a note of that and look into it as I start thinking about QA testing previous integrations and QA tools in general.

Reply:
We definitely do not at all assume that a column has full coverage. Also COMDATE will not work because it will merge everyone's sentence who was committed on the same day. Options:
1) De-duplicate fully and use the person ID as the pl
2) duplicate and use SENTENCE as the pk BUT move the datetime into the association.

I think we've been usually going with option 2.